src/main/java/org/mskcc/oncokb/curation/importer/CdxImporter.java

```diff
@@ -178,70 +176,70 @@ private CompanionDiagnosticDevice parseCdxNameColumn(String columnValue) throws
-                Optional<FdaSubmission> optionalFdaSubmission = fdaSubmissionService.findOrFetchFdaSubmissionByNumber(
-                    number,
-                    supplementNumber,
-                    false
-                );
-                if (optionalFdaSubmission.isPresent()) {
-                    fdaSubmissions.add(optionalFdaSubmission.get());
-                } else {
-                    log.error("Could not find FDA Submission {}", pma);
+    private Optional<FdaSubmission> parseFdaSubmissionColumn(String columnValue) {
+        columnValue = columnValue.trim();
+        Pattern pattern = Pattern.compile("^([A-Za-z\\d]+)(\\/([A-Za-z\\d]+))?\\s*(\\((.*)\\))?");
+        Matcher matcher = pattern.matcher(columnValue);
+        if (matcher.find() && matcher.group(1) != null) {
+            String number = matcher.group(1);
+            String supplementNumber = Optional.ofNullable(matcher.group(3)).orElse("");
+            Optional<FdaSubmission> optionalFdaSubmission = fdaSubmissionService.findOrFetchFdaSubmissionByNumber(
+                number,
+                supplementNumber,
+                false
+            );
+            if (optionalFdaSubmission.isPresent()) {
+                Instant fdaSubmissionDate = cdxUtils.convertDateToInstant(matcher.group(5));
+                if (fdaSubmissionDate != null) {
+                    optionalFdaSubmission.get().setDecisionDate(fdaSubmissionDate);
+                    optionalFdaSubmission.get().curated(true);
                 }
+                return optionalFdaSubmission;
             } else {
-                log.error("Cannot find FDA Submission {}", pma);
+                log.error("Could not find FDA Submission {}", columnValue);
             }
+        } else {
+            log.error("Cannot find FDA Submission {}", columnValue);
         }
-        return fdaSubmissions;
-    }
-
-    private void parseFdaSubmissionDateColumn(List<FdaSubmission> fdaSubmissions, String columnValue) {
-        String[] dates = columnValue.split("and");
-        for (int i = 0; i < dates.length; i++) {
-            String date = dates[i];
-            date = date.replace("(", "").replace(")", "").trim();
-            Instant fdaSubmissionDate = cdxUtils.convertDateToInstant(date);
-            fdaSubmissions.get(i).setDecisionDate(fdaSubmissionDate);
-        }
+        return Optional.empty();
     }
 
-    private CancerType parseCancerTypeColumn(String columnValue) throws IllegalArgumentException {
-        columnValue = columnValue.split(",")[0].trim();
-        Optional<CancerType> optionalCancerType = cancerTypeService.findOneBySubtypeIgnoreCase(columnValue);
-        if (optionalCancerType.isPresent()) {
-            return optionalCancerType.get();
+    private List<CancerType> parseCancerTypesColumn(String columnValue) throws IllegalArgumentException {
+        List<CancerType> cancerTypes = new ArrayList<>();
+        for (String cancerType : columnValue.split("\\|")) {
```

Comment:
I assume you modified the column in CDx sheet to use pipe when there were multiple CT?

Reply:
Yup.